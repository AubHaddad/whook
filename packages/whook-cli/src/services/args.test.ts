import { describe, it, beforeEach, jest, expect } from '@jest/globals';
import initArgs from './args.js';
import type { LogService } from 'common-services';

describe('initArgs', () => {
  const log = jest.fn<LogService>();

  beforeEach(() => {
    log.mockReset();
  });

  it('should parse args', async () => {
    const args = await initArgs({
      log,
      ARGS: ['whook', 'handler', '--name', 'getPing', '--parameters', '{}'],
    });

    expect({
      logCalls: log.mock.calls.filter(([type]) => !type.endsWith('stack')),
      args,
    }).toMatchSnapshot();
  });
});
