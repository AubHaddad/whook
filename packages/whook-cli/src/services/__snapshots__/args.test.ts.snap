// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initArgs should parse args 1`] = `
Object {
  "args": Object {
    "command": "handler",
    "namedArguments": Object {
      "name": "getPing",
      "parameters": "{}",
    },
    "rest": Array [],
  },
  "logCalls": Array [
    Array [
      "debug",
      "ðŸ›  - Parsed args:",
      Object {
        "command": "handler",
        "namedArguments": Object {
          "name": "getPing",
          "parameters": "{}",
        },
        "rest": Array [],
      },
    ],
  ],
}
`;
